src/dev-server.js

```diff
@@ -17,7 +17,13 @@ const ensure = async (dir, log) => {
   }
 }
 
-async function createContext(esbuildConfig, entryPoints, watchMode, plugins = []) {
+const hasStringArrayContentChanged = (oldList, newList) => {
+  return oldList.length !== newList.length || new Set([].concat(oldList, newList)).size !== oldList.length

```

Comment:
If using mathematical sets, might be worth using the built in `difference()` method for it, which signals intent using math speak nomenclature? 🙇🏽 🙏🏽  

If you're worried about overhead from the multiple constructor calls the function could instead return another function that has the `oldSet` closed over but I think that's probably just my brain being pedantic (pre-optimisation something something evil).


```
const isSpecListIdentical = (oldList, newList) => {
    const [ oldSet, newSet ] = [ new Set(oldList), new Set(newList) ];
    return oldSet.difference(newSet).size === 0;
}
```

Reply:
Ah, nice. That tought me about the new methods here. I'd say symmetricDifference would have been the best method here as it would also detect removal:
```
const hasStringArrayContentChanged = (oldList, newList) => {
  return new Set(oldList).symmetricDifference(new Set(newList)).size > 0
}
```
However I'd rather not use it, since this lib is not compiled and support for this requires node 22+.
I guess if I set up TS at some point I might add this in though.

Also feel free to add any reviews/fixes, this lib is very roughly maintained after all. ^^''
At least I'll be quick to react.

(Also I'll close this PR, I already sneakily put everything on the master branch and released and just forgot to close this one)